src/go/k8s/pkg/console/sasl.go

```diff
@@ -143,7 +149,7 @@ func (k *KafkaSA) Cleanup(ctx context.Context) error {
 		return err
 	}
 
-	if err := adminAPI.DeleteUser(ctx, GenerateSASLUsername(k.consoleobj)); err != nil {
+	if err := adminAPI.DeleteUser(ctx, PrefixUsername(k.consoleobj.GetName())); err != nil {
```

Comment:
hmm I know we're rushing but from future maintenancce perspective the username should ideally be defined in one place including the prefixes. What about exposing a method from superusers resource that returns the prefixed one? You can rename the thing you inject into superusers to "usernameRoot" or something likee that and expose method `getUsername`

WDYT?

Reply:
Agree with you. Most of this should go away when we set the console user as superuser. Updated it.